go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -748,7 +749,11 @@ type poolConn interface {
 
 func (qre *QueryExecutor) execSQL(conn poolConn, sql string, wantfields bool) (*sqltypes.Result, error) {
 	defer qre.logStats.AddRewrittenSQL(sql, time.Now())
-	return conn.Exec(qre.ctx, sql, int(qre.tsv.qe.maxResultSize.Get()), wantfields)
+	res, err := conn.Exec(qre.ctx, sql, int(qre.tsv.qe.maxResultSize.Get()), wantfields)
+	if res != nil && int64(len(res.Rows)) > qre.tsv.qe.warnResultSize.Get() {
+		log.Warningf("Results returned exceeds warning threshold: %q", sql)
```

Comment:
qre.ctx is passed through here. It will have the caller id. I'm thinking we should do caller id logging uniformly for all log messages. If so, a utility function in tabletenv package would be the way to go.

As for the `/debug/vars`, we can start a top level variable called "Warnings" (similar to InternalErrors), and report this as "ResultsExceeded" category. There are other places where we issue warnings. We can consolidate those under this.